crates/bevy_asset/src/loader.rs

```diff
@@ -307,6 +308,271 @@ pub enum DeserializeMetaError {
+                    .load_context
+                    .asset_server
+                    .get_path_asset_loader(path)
+                    .await
+                    .map_err(|error| LoadDirectError {
+                        dependency: path.clone(),
+                        error: error.into(),
+                    })?
+            };
+            let meta = loader.default_meta();
+            (meta, loader, ReaderRef::Borrowed(reader))
+        } else {
+            let (meta, loader, reader) = self
+                .base
+                .load_context
+                .asset_server
+                .get_meta_loader_and_reader(path, self.base.asset_type_id)
+                .await
+                .map_err(|error| LoadDirectError {
+                    dependency: path.clone(),
+                    error,
+                })?;
+            (meta, loader, ReaderRef::Boxed(reader))
+        };
+
+        if let Some(meta_transform) = self.base.meta_transform {
+            meta_transform(&mut *meta);
+        }
+
+        let asset = self
+            .base
+            .load_context
+            .load_direct_internal(path.clone(), meta, &*loader, reader.as_mut())
+            .await?;
+        Ok((loader, asset))
+    }
+
+    /// Loads the asset at the given `path` directly. This is an async function that will wait until the asset is fully loaded before
+    /// returning. Use this if you need the _value_ of another asset in order to load the current asset. For example, if you are
+    /// deriving a new asset from the referenced asset, or you are building a collection of assets. This will add the `path` as a
+    /// "load dependency".
+    ///
+    /// If the current loader is used in a [`Process`] "asset preprocessor", such as a [`LoadTransformAndSave`] preprocessor,
+    /// changing a "load dependency" will result in re-processing of the asset.
+    ///
+    /// [`Process`]: crate::processor::Process
+    /// [`LoadTransformAndSave`]: crate::processor::LoadTransformAndSave
+    pub async fn load<A: Asset>(
+        mut self,
+        path: impl Into<AssetPath<'c>>,
+    ) -> Result<LoadedAsset<A>, LoadDirectError> {
+        self.base.asset_type_id = Some(TypeId::of::<A>());
+        let path = path.into().into_owned();
+        self.load_internal(&path)
+            .await
+            .and_then(move |(loader, untyped_asset)| {
+                untyped_asset.downcast::<A>().map_err(|_| LoadDirectError {
+                    dependency: path.clone(),
+                    error: AssetLoadError::RequestedHandleTypeMismatch {
+                        path,
+                        requested: TypeId::of::<A>(),
+                        actual_asset_name: loader.asset_type_name(),
+                        loader_name: loader.type_name(),
+                    },
+                })
+            })
+    }
+}
+
+/// A builder for directly loading untyped nested assets inside a `LoadContext`.
+pub struct UntypedDirectNestedLoader<'a, 'b, 'c> {
```

Comment:
I really want doc comments for these lifetimes, and ideally better (longer) lifetime names.